go/logger/test_logger.go

```diff
@@ -48,11 +49,13 @@ var _ Logger = (*TestLogger)(nil)
 
 // ctx can be `nil`
 func (log *TestLogger) common(ctx context.Context, lvl logging.Level, useFatal bool, fmts string, arg ...interface{}) {
+	log.Lock()
+	defer log.Unlock()
 	if log.log.Failed() {
-		failReported := atomic.SwapInt32(&(log.failReported), 1)
-		if failReported == 0 {
+		if !log.failReported {
```

Comment:
log.Lock could be inside this block

Reply:
I had it there originally, but I think there's a problem in systests with these "TEST FAILED" messages.  They appear multiple times.  I thought moving the lock up might fix it, but it didn't

Reply:
That they appear multiple times is a shortcoming of the way it's done.

Reply:
ok, thanks.